mass_mailing_partner/models/res_partner.py

```diff
@@ -28,15 +30,15 @@ class ResPartner(models.Model):
     @api.one
     @api.constrains('email')
     def _check_email_mass_mailing_contacts(self):
-        if self.mass_mailing_contacts and not self.email:
+        if self.mass_mailing_contact_ids and not self.email:
             raise ValidationError(
                 _("This partner '%s' is subscribed to one or more "
                   "mailing lists. Email must be assigned." % self.name))
 
     @api.one
-    @api.depends('mass_mailing_contacts')
+    @api.depends('mass_mailing_contact_ids')
```

Comment:
You have to add ```mass_mailing_contact_ids.opt_out``` in order to recompute when one contact change from ```opt_out = False``` to ```opt_out = True```

Use case to reproduce:

1. Go to Agrolait partner form
2. Click on Mailing lists smart button (count = 0)
3. Create a contact, Agrolait will be the partner related with it automatically
4. Return to partner form. Mailing lists smart butoon now shows count = 1
5. Click again on Mailing lists smart button
6. Change opt_out column from False to True
7. Return to partner fomr. Mailing lists smart button shows count = 1, but it should show count = 0

To fix this, change api.depends:
```
@api.depends('mass_mailing_contact_ids', 'mass_mailing_contact_ids.opt_out')
```